### PR TITLE
feat: added github_create_gist function

### PR DESCRIPTION
**Pull request by antropez**
I have added a function that will allow the programmer to create a new Gist with a body description, and title. The programmer can also choose the visibility of the gist before they create it.

### CODE REVIEW

File: github-gist.c [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  log_info("===create-gist===");
+
+  if (!params->description) {
+    log_error("Missing 'description'");
+  }
+  if (!params->title) {
+    log_error("Missing 'title'");
+  }
+  if (!params->contents) {
+    log_error("Missing 'contents'");
```
Comment (LucasMull) [inline github-gist.c:26]: should this return ORCA_MISSING_PARAMETER?

File: github-gist.c [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  /* Create the format string for the payload
+   * TODO:
+   * Allocate buffer big enough, then free it after the request is made
+   * */
+  snprintf(fmt, sizeof(fmt), "(public): \"%s\", (description): \"%s\", (files): { (%s): { (content): \"%s\" }}", params->public,
+                                                                                                                 params->description,
+                                                                                                                 params->title,
+                                                                                                                 params->contents);
+  printf("%s\n", fmt);
```
Comment (LucasMull) [inline github-gist.c:40]: Is this printf() a debug remnant, or is it here for logging purposes?

File: github-gist.c (hunk 1 above)
Comment (antropez) [reply github-gist.c:28]: Taken care of.

File: github-gist.c (hunk 2 above)
Comment (antropez) [reply github-gist.c]: It was a debug remnant, it has been removed.

**Review by LucasMull — APPROVED** [review]
LGTM!

**Pull request merged**